Lib/codeop.py

```diff
@@ -84,9 +84,11 @@ def _maybe_compile(compiler, source, filename, symbol):
     except SyntaxError:
         pass
 
-    # Suppress warnings after the first compile to avoid duplication.
+    # Catch syntax warnings after the first compile
+    # to emit SyntaxWarning at most once.
     with warnings.catch_warnings():
-        warnings.simplefilter("ignore")
+        warnings.simplefilter("error", SyntaxWarning)
```

Comment:
Ignoring *all* warnings was intentional.  Compile also issues DeprecationWarnings.  Narrowing the filter to SyntaxWarning only reintroduces the error of DeprecationWarning being issued thrice, which was fixed by the previous code.  I will write more on the issue.

Reply:
Then maybe the catch warnings need to be put in there 3 times, to reset the warning filter on each subsequent compilations ?